recipes/libnice/all/conanfile.py

```diff
@@ -0,0 +1,114 @@
+import os
+from conan import ConanFile
+from conan.tools.meson import Meson, MesonToolchain
+from conan.tools.files import copy, get, rmdir, rename, chdir, rm
+from conan.tools.layout import basic_layout
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.microsoft import is_msvc_static_runtime
+from conan.errors import ConanInvalidConfiguration
+
+
+class LibniceConan(ConanFile):
+    name = "libnice"
+    homepage = "https://libnice.freedesktop.org/"
+    license = ("MPL-1.1", "LGPL-2.1-only")
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "a GLib ICE implementation"
+    topics = ("ice", "stun", "turn")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "crypto_library": ["openssl", "win32"],
+        "with_gstreamer": [True, False],
+        "with_gtk_doc": [True, False],
+        "with_introspection": [True, False]}
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_gstreamer": False,
+        "with_gtk_doc": False,
+        "with_introspection": False}
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            self.options.crypto_library = "win32"
+        else:
+            self.options.crypto_library = "openssl"
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.os != "Windows" and self.options.crypto_library == "win32":
+            raise ConanInvalidConfiguration(
+                "-o libnice:crypto_library=win32 is not supported on non-Windows")
+        if self.settings.os == "Windows" and self.options.with_gtk_doc:
+            raise ConanInvalidConfiguration(
+                "-o libnice:with_gtk_doc=True is not support on Windows")
+        if is_msvc_static_runtime(self) and self.options["glib"].shared:
```

Comment:
```suggestion
        if is_msvc_static_runtime(self) and self.validate["glib"].options.shared:
```

Reply:
Shall it be .dependencies rather than .validation? I couldn't find it in the doc.

Reply:
Yes you are correct!

```suggestion
        if is_msvc_static_runtime(self) and self.dependencies["glib"].options.shared:
```